src/lib/frontend/d_cnf.ml

```diff
@@ -786,7 +786,130 @@ let mk_add translate sy ty l =
     | [] -> []
   in
   let args = aux_mk_add l in
-  E.mk_term sy args ty
+  E.mk_term (Sy.Op Sy.Plus) (List.fast_sort E.compare args) ty
```

Comment:
Isn't this re-introducing the issues we had with #505? Why is it OK to sort the arguments to `+` here?

Reply:
Not intentional, probably a rebase error, I'll fix it.